frontend/src/components/LeadModal/LeadModal.js

```diff
@@ -0,0 +1,388 @@
+  },
+}));
+
+export const LeadModal = ({ open, onClose, addLead }) => {
+  const classes = useStyles();
+  const [assigned, setAssigned] = useState('');
+  const [companyName, setCompanyName] = useState('');
+  const [contactName, setContactName] = useState('');
+  const [email, setEmail] = useState('');
+  const [phone, setPhone] = useState('');
+  const [website, setWebsite] = useState('');
+  const [facebook, setFacebook] = useState('');
+  const [instagram, setInstagram] = useState('');
+  const [linkedin, setLinkedin] = useState('');
+  const [twitter, setTwitter] = useState('');
+  const [users, setUsers] = useState([]);
+
+  const clearForm = () => {
+    setAssigned('');
+    setCompanyName('');
+    setContactName('');
+    setEmail('');
+    setPhone('');
+    setWebsite('');
+    setFacebook('');
+    setInstagram('');
+    setLinkedin('');
+    setTwitter('');
+  };
+
+  const checkForErrors = (response) => {
+    if (response.status === 200) {
+      return response.json();
+    } else {
+      throw new Error('Something went wrong');
+    }
+  };
+
+  useEffect(() => {
+    const getUsers = async () => {
+      const token = localStorage.getItem('partnerFinderToken');
+      const headers = {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${token}`,
+      };
+
+      const response = await fetch(`${API_HOST}/users`, {
+        headers: headers,
+      });
+      const data = await checkForErrors(response);
+      setUsers(data.users);
+    };
+    getUsers();
+  }, []);
+
+  const checkValidPhone = (number) => {
+    const extractedNums = number.replace(/[^0-9]/g, '');
+    if (extractedNums.length !== 10) {
+      return false;
+    } else {
+      return extractedNums;
+    }
+  };
+
+  const checkValidEmail = (email) => {
+    const validEmail = new RegExp('^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$');
+    return validEmail.test(email);
+  };
+
+  const checkValidUrl = (url) => {
+    var validUrl = new RegExp(
+      '^(https?:\\/\\/)?' + // protocol
+        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+        '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+        '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+        '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+        '(\\#[-a-z\\d_]*)?$',
+      'i'
+    );
+    return validUrl.test(url);
+  };
+
+  const checkAssignedUserExists = (assignedUser) => {
+    if (users.includes(assignedUser)) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  const resetErrors = () => {
+    let element;
+    const ids = [
+      'companyNameValidation',
+      'phoneValidation',
+      'fbValidation',
+      'instagramValidation',
+      'linkedinValidation',
+      'twitterValidation',
+      'websiteValidation',
+      'emailValidation',
+      'userValidation',
+    ];
+    ids.forEach((id) => {
+      element = document.getElementById(id);
+      element.classList.add('hidden');
```

Comment:
This might be too little too late here, but might be worth relying on something like https://formik.org/ if we intend to create more forms of this size.

Otherwise, we could do something like the following if we want to avoid direct DOM manipulation:

```
const [formErrors, setFormErrors] = useState({
  companyNameValidation: false,
})

// down below in the return
return (
   ...
   {formErrors.companyNameValidation && <span>Company Name is require</span>}
   <label>Company Name</label>
   ...
)
```

Could also create an `<Input />` component that takes a few props:

```
<Input hasError={formErrors.companyNameValidation} placeholder="lorem ipsum" errorMsg="lorem ipsum" />
```

Reply:
Love these suggestions. I have removed any direct DOM manipulation with these suggestions but am having some difficulty getting the form to behave, I wonder if it has to do with how I am changing the state of the errors? Are these overwriting each other?

```
if (facebook && !checkValidUrl(facebook)) {
      setFormErrors({ ...formErrors, fbValidation: true });
    }
    if (instagram && !checkValidUrl(instagram)) {
      setFormErrors({ ...formErrors, instagramValidation: true });
    }
    if (linkedin && !checkValidUrl(linkedin)) {
      setFormErrors({ ...formErrors, linkedinValidation: true });
    }
```

Reply:
More specifically, I am trying to submit a form where every input should throw an error and I'm not consistently seeing error messages and sometimes the form submits anyway... quite confusing.

Reply:
I wonder if this might be due to async nature of `setState` 🤔

Perhaps something like:

```javascript
const validateInputs = () => {
    resetErrors();
    const errors = {};

    if (!companyName) {
      errors.companyNameValidation = true;
    }
    
    // repeat for other errors

    setFormErrors(errors)
```

Reply:
That was exactly it!